hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/ProcessorTaskletBase.java

```diff
@@ -17,46 +17,41 @@
 package com.hazelcast.jet.impl.execution;
 
 import com.hazelcast.jet.Inbox;
+import com.hazelcast.jet.Outbox;
 import com.hazelcast.jet.Processor;
 import com.hazelcast.jet.Processor.Context;
-import com.hazelcast.jet.impl.util.ArrayDequeOutbox;
 import com.hazelcast.jet.impl.util.CircularListCursor;
 import com.hazelcast.jet.impl.util.ProgressState;
 import com.hazelcast.jet.impl.util.ProgressTracker;
 import com.hazelcast.util.Preconditions;
 
-import javax.annotation.Nonnull;
 import java.util.ArrayDeque;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map.Entry;
 import java.util.Optional;
 import java.util.Queue;
 import java.util.TreeMap;
-import java.util.stream.Stream;
 
-import static com.hazelcast.jet.impl.util.DoneItem.DONE_ITEM;
 import static java.util.Comparator.comparing;
 import static java.util.stream.Collectors.groupingBy;
 import static java.util.stream.Collectors.toCollection;
 
-public class ProcessorTasklet implements Tasklet {
+abstract class ProcessorTaskletBase implements Tasklet {
 
+    final ProgressTracker progTracker = new ProgressTracker();
+    final Processor processor;
+    final OutboundEdgeStream[] outstreams;
+    InboundEdgeStream currInstream;
+
```

Comment:
Wouldn't it be clearer, if these fields are `protected`? It more clearly states, that they are meant to be used in subclasses, not by classes in the same pkg.